src/components/Sender.tsx

```diff
@@ -30,78 +33,112 @@ interface Props {
   } = useSourceTarget();
+  const { setReceiver } = useReceiver();
 
   useEffect(() => {
     if (!chains.length) {
-      setChains([sourceChain, targetChain]);
+      setChains([
+        [sourceChain, targetChain],
+        [targetChain, sourceChain]
+      ]);
     }
   }, [chains.length, sourceChain, targetChain]);
 
   const value = account ? encodeAddress(account.address, getChainSS58(sourceChain)) : '';
 
   const onChange = (value: string, chain: string) => {
     setCurrentAccount(value, chain);
+    setReceiver(null);
   };
-  const renderAccounts = (chain: string) => {
-    const formatedAccounts = formatAccounts(accounts, chain);
+
+  const renderAccounts = (chains: string[]) => {
+    const [source, target] = chains;
+    const formatedAccounts = formatAccounts(accounts, source);
     const items = formatedAccounts.map(({ text, value, key }: any) => (
       <MenuItem
         key={key}
         value={value}
         onClick={() => {
-          onChange(value, chain);
+          onChange(value, source);
         }}
       >
-        <Account text={text} value={value} showDerivedBalance chain={chain} />
+        <div className={classes.row}>
+          <Account accountName={text} value={value} chain={source} />
+          <Account accountName={text} value={value} isDerived fromSender chain={target} />
+        </div>
       </MenuItem>
     ));
-    return [<SubHeader key={chain} chain={chain} />, items];
+    return [<SubHeader key={source} chain={source} />, items];
   };
 
+  const getName = (account: AccountType) => (account!.meta.name as string).toLocaleUpperCase();
+
   const AccountSelected = () => {
     if (account) {
-      const text = (account.meta.name as string).toLocaleUpperCase();
-      return <Account text={text} value={value} />;
+      const text = getName(account);
+      return <Account accountName={text} value={value} chain={sourceChain} />;
     }
     return null;
   };
 
   return (
     <Container className={className}>
-      <InputLabel id="demo-simple-select-label">{sourceChain} Account</InputLabel>
-      <FormControl variant="outlined" className="formControl">
-        <Select
-          value={value}
-          name="name"
-          inputProps={{
-            id: 'name-native-error'
-          }}
-          renderValue={(): React.ReactNode => <AccountSelected />}
-        >
-          {chains.map((chain) => renderAccounts(chain))}
-        </Select>
-      </FormControl>
-      {derivedAccount && (
-        <div className="formControl">
-          <Account text="Derived Account" value={derivedAccount} />
-        </div>
-      )}
+      <InputLabel className="label" id="demo-simple-select-label">
+        {sourceChain} Account
+      </InputLabel>
+      <div className="senderSelect">
+        <FormControl className="formControl">
+          <Select value={value} name="name" renderValue={(): React.ReactNode => <AccountSelected />}>
+            {chains.map((chain) => renderAccounts(chain))}
```

Comment:
Your code has an issue and this is why i reverted it to the original one: 

When you flip the chains because a target account was selected, the order in the dropdown is flipped as well.